### PR TITLE
Remove calls to pprint() from in use catalog endpoints

### LINKED ISSUE #1452: Remove calls to pprint() from in use catalog endpoints
Mentioned by @hornc in https://github.com/internetarchive/openlibrary/pull/1317#commitcomment-30872900

### PR DESCRIPTION
**Pull request by YBthebest**
> **Description**: What does this PR achieve? [feature|hotfix|refactor]

Fixes #1452 by removing some calls to pprint()

> **Technical**: What should be noted about the implementation?

> **Testing**: Steps for reviewer to reproduce / verify this PR fixes the problem?

Search through the project for usage of pprint to see if the remaining instances are correctly remaining, and check in the commit if the removed/modified instances are correctly removed.

### CODE REVIEW

File: openlibrary/catalog/works/find_works.py [hunk 1]
```diff
@@ -535,8 +535,6 @@ def new_work(akey, w, do_updates, fh_log):
         try:
             print(ol.save_many(update, "add editions to new work"), file=fh_log)
         except:
```
Comment (mekarpeles) [inline openlibrary/catalog/works/find_works.py]: This needs to raise!

**Comment by mekarpeles** [comment]
```
python -m flake8 . --count --exclude=./.*,scripts/20*,vendor/*  --select=E901,E999,F822,F823 --show-source --statistics
./openlibrary/catalog/works/find_works.py:538:14: E999 IndentationError: expected an indented block
        return [wkey]
```

File: openlibrary/catalog/works/find_works.py (hunk 1 above)
Comment (cclauss) [reply openlibrary/catalog/works/find_works.py]: Or remove the try and except lines and dedent line 536.  Bare exceptions blow in any case.  
https://realpython.com/the-most-diabolical-python-antipattern/

File: openlibrary/catalog/works/find_works.py [hunk 2]
```diff
@@ -598,8 +596,6 @@ def update_work_with_best_match(akey, w, work_to_edition, do_updates, fh_log):
         if cur_work['title'] != w['title']:
             print(( 'update work title:', best, repr(cur_work['title']), '->', repr(w['title'])))
         existing_work = get_with_retry(best)
-        if existing_work['type'] != '/type/work':
-            pprint(existing_work)
         assert existing_work['type'] == '/type/work'
```
Comment (cclauss) [inline openlibrary/catalog/works/find_works.py]: Help the developer to understand failures by adding a string to the end of the assert on line 599
* assert existing_work[‘type’] == ‘/type/work’, “{type} == ‘/type/work’”.format(**existing_work)

File: openlibrary/catalog/works/find_works.py [hunk 3]
```diff
@@ -750,8 +745,6 @@ def other_matches(w, existing_wkey):
     works_updated_this_session = set()
 
     for w in works: # 4th pass
-        if 'has_conflict' in w:
-            pprint(w)
         assert 'has_conflict' not in w
```
Comment (cclauss) [inline openlibrary/catalog/works/find_works.py]: Help the developer to understand failures by adding a string to the end of the assert on line 748
* assert ‘has_conflict’ not in w, ‘w: {}’.format(w)

File: openlibrary/catalog/works/find_works.py (hunk 1 above)
Comment (YBthebest) [reply openlibrary/catalog/works/find_works.py]: That does make more sense, thanks for the tip.

File: openlibrary/catalog/author/rename.py [hunk 4]
```diff
@@ -211,8 +207,6 @@ def merge_authors(author, merge_with, name):
             print("date mismatch")
             continue
     except KeyError:
-        pprint(author)
-        pprint(merge_with)
```
[the remark this answers is not in the document]
Comment (YBthebest) [reply openlibrary/catalog/author/rename.py]: Good question, wonder how I missed that one. Thanks!

File: openlibrary/catalog/marc/db/web_marc_db.py [hunk 5]
```diff
@@ -4,8 +4,7 @@
 from catalog.get_ia import get_data
 from catalog.marc.build_record import build_record
 from catalog.marc.fast_parse import get_all_subfields, get_tag_lines, get_first_tag, get_subfields
-from openlibrary.catalog.utils import cmp
```
[the remark this answers is not in the document]
Comment (YBthebest) [reply openlibrary/catalog/marc/db/web_marc_db.py]: Definitely a mistake while I was removing the unused pprint imports!

File: openlibrary/catalog/works/find_works.py (hunk 2 above)
Comment (YBthebest) [reply openlibrary/catalog/works/find_works.py]: That is a neat little construct, thanks for the tip

**Comment by YBthebest** [comment]
@cclauss @mekarpeles thanks for your very helpful tips. I made the necessary modifications to the GitHub issue as well as the requested changes to the code, all of which made a lot of sense. Let me know if there is anything left!

**Comment by cclauss** [comment]
Eeeekk!  I am not a fan of the last commit.  Is it possible to roll it back and make those changes on _separate_ PRs?  This changes have nothing to do with __pprint()__ so they do not belong on this PR.  They will only confuse reviewers.

**Comment by YBthebest** [comment]
Yes @cclauss I know, I noted this on the related GitHub issue, my bad, it wasn't on purpose.

**Comment by cclauss** [comment]
Understood...  Git can be difficult in the early daze.

**Comment by YBthebest** [comment]
Ok, I reverted the changes, and we're now back to the old state again. Sorry about that once again !

**Comment by cclauss** [comment]
Nice work @YBthebest 👍

**Pull request merged**